packages/snaps-ui/src/validation.ts

```diff
@@ -25,3 +25,72 @@ export function isComponent(value: unknown): value is Component {
 export function assertIsComponent(value: unknown): asserts value is Component {
   assertStruct(value, ComponentStruct, 'Invalid component');
 }
+
+const LINK_REGEX = /(?<protocol>[a-z]+:\/?\/?)(?<host>\S+?(?:\.[a-z]+)+)/giu;
+
+enum AlloweProtocols {
+  Https = 'https:',
+  Mailto = 'mailto:',
+}
+
+const ALLOWED_PROTOCOLS = Object.values(AlloweProtocols);
+
+/**
+ * Searches for links in a sting and checks them against the phishing list.
+ *
+ * @param text - The text to verify.
+ * @param isOnPhishingList - The function that checks the link against the phishing list.
+ */
+export function assertLinksAreSafe(
+  text: string,
+  isOnPhishingList: (url: string) => boolean,
+) {
+  const links = text.match(LINK_REGEX);
+  if (links) {
+    links.forEach((link) => {
+      try {
+        const url = new URL(link);
+        assert(
+          ALLOWED_PROTOCOLS.includes(url.protocol as AlloweProtocols),
+          `protocol must be one of: ${ALLOWED_PROTOCOLS.join(', ')}`,
+        );
+
+        const hostname =
+          url.protocol === AlloweProtocols.Mailto
+            ? url.pathname.split('@')[1]
+            : url.hostname;
+
+        assert(!isOnPhishingList(hostname), 'detected as phishing');
+      } catch (error) {
+        throw new Error(
+          `Invalid URL: ${
+            error instanceof AssertionError ? error.message : 'invalid sintax'
```

Comment:
```suggestion
            error instanceof AssertionError ? error.message : 'Invalid syntax.'
```

When is this thrown?

Reply:
`Invalid syntax` is thrown if there's an error with `new URL()`

Reply:
Maybe "Unable to parse URL."?